src/material/autocomplete/autocomplete-trigger.ts

```diff
@@ -480,6 +480,14 @@ export class MatAutocompleteTrigger implements ControlValueAccessor, AfterViewIn
 
   /** If the label has been manually elevated, return it to its normal state. */
   private _resetLabel(): void  {
+    const value: any = this._formField
+      && this._formField._control.value
+      || this._element.nativeElement.value;
+
+    if (value != '' && value != null) {
```

Comment:
This check shouldn't be inside `_resetLabel` but only around the relevant call site.